lib/coll.gd

```diff
@@ -2400,6 +2400,38 @@ DeclareOperation( "AsSSortedListNonstored", [IsListOrCollection] );
 DeclareGlobalFunction( "Elements" );
 
 
+#############################################################################
+##
+##  TODO: document FoldLeft
+##
+DeclareGlobalFunction( "FoldLeft" );
+
+
+#############################################################################
+##
+##  TODO: document FoldLeftOp
+##
+DeclareOperation( "FoldLeftOp", [ IsListOrCollection, IsFunction ] );
+
+
+#############################################################################
+##
+##  TODO: document FoldLeftX
+##
+DeclareGlobalFunction( "FoldLeftX" );
+
+
+#############################################################################
+##
+##  TODO: document the following
+##
+DeclareGlobalFunction( "ForAllX" );
+DeclareGlobalFunction( "ForAnyX" );
+DeclareGlobalFunction( "FilteredX" );
+DeclareGlobalFunction( "NumberX" );
+DeclareGlobalFunction( "PerformX" );
```

Comment:
TODO: also add `FirstX` and `LastX`?

Reply:
The "problem" with `FirstX` is the same as for `ForAnyX` and `ForAllX`: we want to be able to short circuit and abort the search as soon as we know the answer.  The mechanism for doing this in this PR is the ugliest part (IMHO) of this PR by far (there is an optional `abortValue` that one can specify, and for which we check by identity comparison... once encountered, the search aborts. Works, is simply, but... ugly?